### ISSUE TITLE
Block Widgets: Cookie & Consents Banner Widget - Unexpected error if added on the Free plan 

### ISSUE THREAD

**Issue by wpshellbelle** [Automattic/wp-calypso#56292]
### Quick summary

When adding the Cookies and Consent Banner using the new block widgets, you will see the following error upon adding this widget to a site on a Free WordPress.com plan: `The editor has encountered an unexpected error.`

### Steps to reproduce

1. On a WordPress.com site on the Free plan, go to Appearance > Customize > Widgets > [Widget Area] > + Add a Widget
2. Add the Cookies and Consents Banner block widget (must be using block widgets to duplicate error) 
3. See error `The editor has encountered an unexpected error.` and *Copy Error* 

**Error:**

_createWidget@https://s0.wp.com/wp-content/plugins/gutenberg-core/v11.4.1/build/customize-widgets/index.min.js?m=1630518223h&ver=546484628ba943033ba6605bfed8f168:9:12626
_updateWidgets/n<@https://s0.wp.com/wp-content/plugins/gutenberg-core/v11.4.1/build/customize-widgets/index.min.js?m=1630518223h&ver=546484628ba943033ba6605bfed8f168:9:13713
_updateWidgets@https://s0.wp.com/wp-content/plugins/gutenberg-core/v11.4.1/build/customize-widgets/index.min.js?m=1630518223h&ver=546484628ba943033ba6605bfed8f168:9:13619
setWidgets@https://s0.wp.com/wp-content/plugins/gutenberg-core/v11.4.1/build/customize-widgets/index.min.js?m=1630518223h&ver=546484628ba943033ba6605bfed8f168:9:13908
r</<@https://s0.wp.com/wp-content/plugins/gutenberg-core/v11.4.1/build/customize-widgets/index.min.js?m=1630518223h&ver=546484628ba943033ba6605bfed8f168:9:3067
ma@https://s0.wp.com/wp-content/plugins/gutenberg-core/v11.4.1/vendor/react-dom.min.edd8b7a6.js?m=1624062436h:7:8328
lc@https://s0.wp.com/wp-content/plugins/gutenberg-core/v11.4.1/vendor/react-dom.min.edd8b7a6.js?m=1624062436h:7:8781
useState@https://s0.wp.com/wp-content/plugins/gutenberg-core/v11.4.1/vendor/react-dom.min.edd8b7a6.js?m=1624062436h:10:5919
c.useState@https://s0.wp.com/wp-content/plugins/gutenberg-core/v11.4.1/vendor/react.min.755b0ae9.js?m=1624062436h:5:7080
@https://s0.wp.com/wp-content/plugins/gutenberg-core/v11.4.1/build/customize-widgets/index.min.js?m=1630518223h&ver=546484628ba943033ba6605bfed8f168:9:2458
xe@https://s0.wp.com/wp-content/plugins/gutenberg-core/v11.4.1/build/customize-widgets/index.min.js?m=1630518223h&ver=546484628ba943033ba6605bfed8f168:9:3227
Le@https://s0.wp.com/wp-content/plugins/gutenberg-core/v11.4.1/vendor/react-dom.min.edd8b7a6.js?m=1624062436h:7:7581
Re@https://s0.wp.com/wp-content/plugins/gutenberg-core/v11.4.1/vendor/react-dom.min.edd8b7a6.js?m=1624062436h:7:14438
Pj@https://s0.wp.com/wp-content/plugins/gutenberg-core/v11.4.1/vendor/react-dom.min.edd8b7a6.js?m=1624062436h:10:10973
di@https://s0.wp.com/wp-content/plugins/gutenberg-core/v11.4.1/vendor/react-dom.min.edd8b7a6.js?m=1624062436h:8:5474
Nj@https://s0.wp.com/wp-content/plugins/gutenberg-core/v11.4.1/vendor/react-dom.min.edd8b7a6.js?m=1624062436h:8:5405
sc@https://s0.wp.com/wp-content/plugins/gutenberg-core/v11.4.1/vendor/react-dom.min.edd8b7a6.js?m=1624062436h:8:5263
gf@https://s0.wp.com/wp-content/plugins/gutenberg-core/v11.4.1/vendor/react-dom.min.edd8b7a6.js?m=1624062436h:8:2127
$g/<@https://s0.wp.com/wp-content/plugins/gutenberg-core/v11.4.1/vendor/react-dom.min.edd8b7a6.js?m=1624062436h:6:12524
unstable_runWithPriority@https://s0.wp.com/wp-content/plugins/gutenberg-core/v11.4.1/vendor/react.min.755b0ae9.js?m=1624062436h:5:3609
Za@https://s0.wp.com/wp-content/plugins/gutenberg-core/v11.4.1/vendor/react-dom.min.edd8b7a6.js?m=1624062436h:6:12302
$g@https://s0.wp.com/wp-content/plugins/gutenberg-core/v11.4.1/vendor/react-dom.min.edd8b7a6.js?m=1624062436h:6:12472
ja@https://s0.wp.com/wp-content/plugins/gutenberg-core/v11.4.1/vendor/react-dom.min.edd8b7a6.js?m=1624062436h:6:12405
@https://s0.wp.com/wp-content/plugins/gutenberg-core/v11.4.1/vendor/react-dom.min.edd8b7a6.js?m=1624062436h:10:14674
Ki@https://s0.wp.com/wp-content/plugins/gutenberg-core/v11.4.1/vendor/react-dom.min.edd8b7a6.js?m=1624062436h:5:1411

### What you expected to happen

I did not expect to see the Cookies and Consents Banner available on the Free plan, [this should require a paid plan. ](https://wordpress.com/support/widgets/cookie-widget/)

### What actually happened

Widget was available but shows error when added. 

### Context

User reported at 30787970-hc

**Comment by Robertght** [comment]
Thank you for reporting this @wpshellbelle ! I can confirm  that I see the same thing happened on my end:

```
TypeError: Cannot read properties of undefined (reading 'get')
    at Oe._createWidget (https://s0.wp.com/wp-content/plugins/gutenberg-core/v11.4.1/build/customize-widgets/index.min.js?m=1630518223h&ver=546484628ba943033ba6605bfed8f168:9:12628)
    at https://s0.wp.com/wp-content/plugins/gutenberg-core/v11.4.1/build/customize-widgets/index.min.js?m=1630518223h&ver=546484628ba943033ba6605bfed8f168:9:13713
    at Array.map (<anonymous>)
    at Oe._updateWidgets (https://s0.wp.com/wp-content/plugins/gutenberg-core/v11.4.1/build/customize-widgets/index.min.js?m=1630518223h&ver=546484628ba943033ba6605bfed8f168:9:13619)
    at Oe.setWidgets (https://s0.wp.com/wp-content/plugins/gutenberg-core/v11.4.1/build/customize-widgets/index.min.js?m=1630518223h&ver=546484628ba943033ba6605bfed8f168:9:13908)
    at https://s0.wp.com/wp-content/plugins/gutenberg-core/v11.4.1/build/customize-widgets/index.min.js?m=1630518223h&ver=546484628ba943033ba6605bfed8f168:9:3067
    at ma (https://s0.wp.com/wp-content/plugins/gutenberg-core/v11.4.1/vendor/react-dom.min.edd8b7a6.js?m=1624062436h:7:8328)
    at lc (https://s0.wp.com/wp-content/plugins/gutenberg-core/v11.4.1/vendor/react-dom.min.edd8b7a6.js?m=1624062436h:7:8781)
    at Object.useState (https://s0.wp.com/wp-content/plugins/gutenberg-core/v11.4.1/vendor/react-dom.min.edd8b7a6.js?m=1624062436h:10:5919)
    at c.useState (https://s0.wp.com/wp-content/plugins/gutenberg-core/v11.4.1/vendor/react.min.755b0ae9.js?m=1624062436h:5:7080)
```

I think this might happen because the already contains the default banner on the Free plan. Adding this to @Automattic/flow-patrol-create

**Comment by sajmes** [comment]
Another report here: 4318561-zd-woothemes

Workaround to fix this involved disabling block widgets on Blog RC and then removing empty HTML widgets from customizer. Once removed, block widgets worked again once enabled.

**Comment by alshakero** [comment]
It seems there are three parts to the story:

1. The `eu_law_consent_widget` is not registered because these lines (wp-content/mu-plugins/widgets/eu-cookie-law/eu-cookie-law.php?r=228827#348-353) return false. This is what causes the exception in the customizer.
2. Then comes this filter (wp-content/mu-plugins/widgets/eu-cookie-law/eu-cookie-law.php?r=228827#169) and disables the widget in the customizer (which makes sense, but not while you're editing the block itself). 
3. If you jump these two (jump the boolean and comment out the filter) the widget will appear and fade away insantly, but I can't figure out why.

**Comment by stefancln** [comment]
Another report here 4382244-HC

**Comment by arcangelini** [comment]
After spending some time working on another issue related to this. I think this could be fixed by updating this check (wp-content/mu-plugins/widgets/eu-cookie-law/eu-cookie-law.php?r=228827#143).

The widget block editor uses the public API to retrieve the actual widget which means it is not coming from the site. This means that global variables that may be set on the actual blog are not accessible. Since the initial check for existing cookie widgets is not getting the correct value it continues on instead of `return`ing as it should.

Just a thought?

**Comment by arcangelini** [comment]
I was able to make a fix for this using the `widget_types_to_hide_from_legacy_widget_block` filter. I added a method that checks for the current plan and hides the widget from the legacy block if it is on the free plan.

This fix does not _fix the error_ that is produced from trying to add a Cookie Widget to a free plan. Instead, it just _removes the ability to add the widget to free plans_. This is how it was in the past before the block editor I believe and the impression we give in the documentation - https://wordpress.com/support/widgets/cookie-widget/#how-to-enable-the-widget

Created the patch and should be able to apply it this week - D69890-code
